### DISCUSSION TITLE
Get scale of chord

### DISCUSSION THREAD

**Discussion by zipzapduo** [melanchall/drywetmidi#136] [General]
Is it possible to get the scale of a chord ?

**Comment by melanchall** [comment]
Hi,

Can you please provide some examples of chords and their scales, so I can suggest a solution? I Have an assumption what you need, but it can differ fromwhat you actually want (I'm not an expert in music theory).

Thanks,
Max

**Comment by zipzapduo** [comment]
Let's say that I have one chord like F A D . What I want to get is D minor

**Reply by melanchall** [reply]
Can you please exaplain why `F A D` is exactly `D minor`? As I can see `F A D` can be steps of many other scales.

**Reply by zipzapduo** [reply]
Yes it is not the scale, my appologies, but the tone

**Reply by melanchall** [reply]
Maybe you need the **name** of a chord? This code:

```csharp
var chord = new Chord(NoteName.F, NoteName.A, NoteName.D);
var names = chord.GetNames();
```

will give you all names of the chord. In this case it will be:

```
Dm
Dmin
D5/F
```

Names are always sorted by length of a name, so in most cases you can just take first name and it would be what you need:

```csharp
var name = chord.GetNames().FirstOrDefault();
```

**Reply by zipzapduo** [reply]
Many many thanks. It is exactly what I need. I don't know if I should create another topic for this but my main goal here is to transpose all chords from the collection to the first with scale correction. Can I do that with this library?

**Reply by melanchall** [reply]
It would be better to start a new thread in Discussions or create a new issue. And if you have a question, it's better to create discussion in Q&A category so we can mark answer and close the question. I personally prefer issues to handle user problems :)
